src/Components/Components/src/ComponentsActivitySource.cs

```diff
@@ -11,115 +12,71 @@ namespace Microsoft.AspNetCore.Components;
         {
-            if (activity.IsAllDataRequested)
-            {
-                if (_circuitId != null)
-                {
-                    activity.SetTag("aspnetcore.components.circuit.id", _circuitId);
-                }
-                if (httpContext != default)
-                {
-                    activity.AddLink(new ActivityLink(httpContext));
-                }
-            }
-            activity.DisplayName = $"Circuit {circuitId ?? ""}";
+            var httpActivity = Activity.Current;
+            activity.DisplayName = $"Route {route ?? "[unknown path]"} -> {componentType ?? "[unknown component]"}";
+            Activity.Current = null; // do not inherit the parent activity
             activity.Start();
-            _circuitContext = activity.Context;
-        }
-        return activity;
-    }
-
-    public void FailCircuitActivity(Activity? activity, Exception ex)
-    {
-        _circuitContext = default;
-        if (activity != null && !activity.IsStopped)
-        {
-            activity.SetTag("error.type", ex.GetType().FullName);
-            activity.SetStatus(ActivityStatusCode.Error);
-            activity.Stop();
-        }
-    }
-
-    public Activity? StartRouteActivity(string componentType, string route)
-    {
-        if (_httpContext == default)
-        {
-            _httpContext = CaptureHttpContext();
-        }
 
-        var activity = ActivitySource.CreateActivity(OnRouteName, ActivityKind.Internal, parentId: null, null, null);
-        if (activity is not null)
-        {
             if (activity.IsAllDataRequested)
             {
-                if (_circuitId != null)
-                {
-                    activity.SetTag("aspnetcore.components.circuit.id", _circuitId);
-                }
                 if (componentType != null)
                 {
                     activity.SetTag("aspnetcore.components.type", componentType);
                 }
                 if (route != null)
                 {
                     activity.SetTag("aspnetcore.components.route", route);
+
+                    // store self link
+                    _componentsActivityLinkStore!.SetActivityContext(ComponentsActivityLinkStore.Route, activity.Context,
+                        new KeyValuePair<string, object?>("aspnetcore.components.route", route));
                 }
-                if (_httpContext != default)
-                {
-                    activity.AddLink(new ActivityLink(_httpContext));
-                }
-                if (_circuitContext != default)
+                if (httpActivity != null && httpActivity.Source.Name == "Microsoft.AspNetCore")
                 {
-                    activity.AddLink(new ActivityLink(_circuitContext));
+                    // store the http link
+                    _componentsActivityLinkStore!.SetActivityContext(ComponentsActivityLinkStore.Http, httpActivity.Context, null);
                 }
             }
```

Comment:
We don't need this bit, do we? It's already set in `RazorComponentEndpointInvoker` and `CircuitHost/ComponentHub`

Reply:
yes, thanks! removed